script.js

```diff
@@ -1,12 +1,261 @@
+    selectShowTag.addEventListener("click", showSelection);
+    navForm.appendChild(selectShowTag);
+    selectAllShows.addEventListener("click", showSelection);
+
+    function showSelection() {
+      if (showOptionTag.value === selectShowTag.value) {
+        apiLink = show.id;
+        api = `https://api.tvmaze.com/shows/${apiLink}/episodes`;
+        getShowDataFromApi(api);
+      }
+    }
+  })
+
+  function getEpisodeFromApi(api) {
+    fetch(api)
+    .then((response) => {
+      if (response.status >= 200 && response.status <= 299) {
+        return response.json();
+      } else {
+        throw new Error (
+          `Encountered an error: ${response.status} ${response.statusText}`
+        );
+      }
+    })
+    .then((allShows) => {
+      makePageForShows(allShows);
+    })
+    .catch((error) => {
+      console.log(error);
+    });
+  }
+
+  function getShowDataFromApi(api) {
+    fetch(api)
+    .then((response) => {
+      if (response.status >= 200 && response.status <= 299) {
+        return response.json();
+      } else {
+        throw new Error (
+          `Encountered an error: ${response.status} ${response.statusText}`
+        );
+      }
+    })
+    .then((allEpisodes) => {
+      makePageForEpisodes(allEpisodes);
+    })
+    .catch((error) => {
+      console.log(error);
+    });
+  }
 }
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+function makePageForShows(allShows) {
+
+  // Search bar
+  const searchBar = document.getElementById("searchBar");
+  searchBar.className = "form-control col-3 ml-4 mt-2 mr-2 mb-3";
+  searchBar.addEventListener("keyup", (event) => {
+    const searchString = event.target.value.toLowerCase();
+    const showCards = document.querySelectorAll(".show-card");
+    let countShows = 0;
+
+    // Checks if each card contains the search string and if not sets display to none
+    for (let i = 0; i < showCards.length; i++) {
+      if (!showCards[i].innerHTML.toLowerCase().includes(searchString)) {
+        showCards[i].style.display = "none";
+      } else {
+        showCards[i].style.display = "block";
+        countShows ++;
+      }
+    }
+    counter.textContent = `Displaying ${countShows}/240 show(s)`;
+  });
+
+  allShows.forEach((show) => {
+
+    // Hides all show cards and displays only the show selected
+    function showDisplay() {
+      showCard.style.display = "none";
+      if (showOptionTag.value === selectShowTag.value) {
+        showCard.style.display = "block";
+        counter.textContent = `Displaying 1/ 240 show(s)`;
+      }
+      // Displays all cards when "All Episodes" selected
+      if (api === 'https://api.tvmaze.com/shows') {
+        showCard.style.display = "block";
+        counter.textContent = `Displaying ${allShows.length}/240 show(s)`;
+      }
+    }
+    // showDisplay();
```

Comment:
I have noticed you created a function and never called it. I just noticed you commented out that function.